File: Client Applications/Ruby Client/track_search.rb
require './util'
require './settings'
require 'open-uri'
require 'rubygems'
require 'json'

def beginTrackSearch(trackTitle, applicationID, applicationKey)

	# Set the request URL
	url = $apiAddress + 'Tracks/?trackTitle=' + URI::encode(trackTitle) + '&format=json'
		
	# Send the GET request
	resp = request(url, applicationID, applicationKey)

	# Interpret the JSON response 
	data = JSON.parse(resp.body)

	# Return the collection of track objects
	return data['ResultSet']
	
end

# Search for tracks by name
puts 'Please enter a track name:'
trackName = gets.chomp
result = beginTrackSearch(trackName, $applicationID, $applicationKey)

# Output the search results
result.each { 
	|track| puts track['Name'] + ' - ' + track['Artists']
}

puts 'Press Enter to continue...'
input = gets.chomp

File: Client Applications/Python Client/track_search.py
import util
import json
import urllib
import settings

def beginTrackSearch(trackTitle, applicationID, applicationKey):

	# Set the request URL
	url = settings.apiAddress + 'Tracks/?trackTitle=' + urllib.parse.quote_plus(trackTitle) + '&format=json'

	# Send the GET request
	resp = util.request(url, applicationID, applicationKey)

	# Interpret the JSON response 
	data = json.loads(resp.decode('utf8'))

	# Return the collection of track objects
	return data['ResultSet']

# Search for tracks by name	
trackName = input('Please enter a track name:')
result = beginTrackSearch(trackName, settings.applicationID, settings.applicationKey)

# Output the search results
for track in result:
	print(track['Name'] + ' - ' + track['Artists'])

input('Press Enter to continue...')
